### PR TITLE
Add example for drawing an error band around a curve

### LINKED ISSUE #17398: Plotting an error band along a curve
### Bug report

**Bug summary**

pyplot.fill_between(x, y1, y2) does not fill areas, where curves y1 and y2 are crossing.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
import numpy as np

def lissajous(t):
    """Calculate Lissajous Curve"""
    x = np.cos(t)
    y = np.sin(2 * t)
    return x, y

data = np.arange(0, 2 * np.pi, 0.01)

plt.plot(*lissajous(data))
plt.fill_between(
    lissajous(data)[0], lissajous(data)[1] - 0.5, lissajous(data)[1] + 0.5, alpha=0.25
)

plt.show()
```

**Actual outcome**

Lissajous_issue

**Expected outcome**

Lissajous_expected

**Note**
I came across this problem while trying to use fill_between() as 'errorbars' for parametric functions. If this is not the intended use, I would be glad if someone has suggestions for a workaround.
Keep up the good work!

**Matplotlib version**

  * Operating system: Arch Linux x86_64, Kernel 5.6.11-arch1-1
  * Matplotlib version: 3.1.3
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.7.7
  * Jupyter version (if applicable):
  * Other libraries: numpy: 1.18.1

matplotlib installed via anaconda

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Inspired by and closes #17398.

### CODE REVIEW

**Comment by timhoffm** [comment]
> ModuleNotFoundError: No module named 'scipy'

Uh, don't we have scipy for the examples?

**Comment by QuLogic** [comment]
No, only the double pendulum example uses it, and it's sgskip.

**Comment by tacaswell** [comment]
Hmm, I also thought we had a docs dependency on scipy, but apparently not, we merged #9401 in 2017 which removed scipy from the doc requirements.

**Comment by QuLogic** [comment]
This will be available after #17477 is merged.

File: examples/lines_bars_and_markers/curve_error_band.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#############################################################################
+# An error band can be used to indicate the uncertainty of the the curve.
+# In this example we assume that the error can be given as a scalar *err*
+# that describes the uncertainty perpendicular to the curve in every point.
+#
+# We visualize this error as a colored band around the path using a
+# `.PathPatch`. The patch is created from two path segments *(xp, yp)*, and
+# *(xn, yn)* that are shifted by +/- *err* perpendicular to the curve *(x, y)*.
+#
+# Note: This method of using a `.PathPatch` is suited for arbitrary curves in
```
Comment (story645) [inline examples/lines_bars_and_markers/curve_error_band.py:37]: suited to (tried to just do the commit suggestion but github freaked out on me)

File: examples/lines_bars_and_markers/curve_error_band.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+phi = np.linspace(0, 2 * np.pi, N)
+r = 0.5 + np.cos(phi)
+x, y = r * np.cos(phi), r * np.sin(phi)
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+plt.show()
+
+#############################################################################
+# An error band can be used to indicate the uncertainty of the the curve.
```
Comment (QuLogic) [inline examples/lines_bars_and_markers/curve_error_band.py:29]:
```suggestion
# An error band can be used to indicate the uncertainty of the curve.
```

File: examples/lines_bars_and_markers/curve_error_band.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# We visualize this error as a colored band around the path using a
+# `.PathPatch`. The patch is created from two path segments *(xp, yp)*, and
+# *(xn, yn)* that are shifted by +/- *err* perpendicular to the curve *(x, y)*.
+#
+# Note: This method of using a `.PathPatch` is suited to arbitrary curves in
+# 2D. If you just have a standard y-vs.-x plot, you can use the simpler
+# `~.Axes.fill_between` method (see also
+# :doc:`/gallery/lines_bars_and_markers/fill_between_demo`).
+
+err = 0.05 * np.sin(2*phi)**2 + 0.04 + 0.02 * np.cos(9*phi + 2)
```
Comment (QuLogic) [inline examples/lines_bars_and_markers/curve_error_band.py:42]: This is an arbitrary choice of error, right? Should get a comment saying so.

Comment (timhoffm) [reply examples/lines_bars_and_markers/curve_error_band.py]: Not sure this is necessary. Almost all our examples use arbitrary data. But done.

Comment (QuLogic) [reply examples/lines_bars_and_markers/curve_error_band.py]: Sure, but this example is about adding an error band, so it _might_ appear as if this is something related to adding the curve, and not just a random error.

**Pull request merged**